node/impl/full/eth_utils.go

```diff
@@ -713,7 +720,18 @@ func newEthTxReceipt(ctx context.Context, tx ethtypes.EthTx, lookup *api.MsgLook
 	}
 
 	baseFee := parentTs.Blocks()[0].ParentBaseFee
-	gasOutputs := vm.ComputeGasOutputs(lookup.Receipt.GasUsed, int64(tx.Gas), baseFee, big.Int(tx.MaxFeePerGas), big.Int(tx.MaxPriorityFeePerGas), true)
+
+	gasFeeCap, err := tx.GasFeeCap()
+	if err != nil {
+		return api.EthTxReceipt{}, xerrors.Errorf("failed to get gas fee cap: %w", err)
```

Comment:
Do we want to error here given a legacy transaction?  Not sure who calls this but if we expect legacy txs then it would probably be better to handle explicitly.  I think this might be what youre top level comment is referring to

Reply:
@ZenGround0 This wont error for legacy transactions as it will return the `GasPrice` instead of `MaxFeePerGas`.

```

func (tx *EthTx) GasFeeCap() (EthBigInt, error) {
	if tx.GasPrice == nil && tx.MaxFeePerGas == nil {
		return EthBigInt{}, fmt.Errorf("gas fee cap is not set")
	}
	if tx.MaxFeePerGas != nil {
		return *tx.MaxFeePerGas, nil
	}
	return *tx.GasPrice, nil
}
```

Yeah this is what I am talking about in my top level comment. Setting the `MaxFeePerGas` on legacy txns feels wrong and `go-ethereum` does not do it either.

I am just waiting for feedback from @raulk or someone else with ETH experience and am happy to change it. However, this is not a consensus breaking change and so can easily be changed in a feature release.